### PR TITLE
Added DeleteConfirmationModal to the work activity table

### PR DESCRIPTION
**Pull request by arkadeep**
Hey @mddilley, thank you for your patience on this—the PR is ready for issue #[18324](https://github.com/cityofaustin/atd-data-tech/issues/18324#event-13635944813) I tested it on my local setup, so I think everything is fine, but this being my first time writing any React code, there might be something wrong (hopefully not though). 

## Associated issues
Closes https://github.com/cityofaustin/atd-data-tech/issues/18324

## Testing
**URL to test:** 

https://deploy-preview-1390--atd-moped-main.netlify.app/

**Steps to test:**
1. Delete a work activity row and see the updated delete confirmation modal matches the one in the funding table

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [x] Product manager checked [DB view dependencies](https://atd-dts.gitbook.io/moped-documentation/product-management/updating-the-arcgis-online-components-database-view)
~- [ ] Product manager added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable~

### CODE REVIEW

File: moped-editor/src/views/projects/projectView/ProjectWorkActivity/ProjectWorkActivityTable.js
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectWorkActivity/ProjectWorkActivityTable.js:163]: Could you remove this comment and the ones on lines 166, 192, and 209? You might have planned to do this already before merge but I'd like to limit comments to ones that help future developers understand the intent of the code. I think that your naming of the functions and state variables does a great job of that on their own. 💯

File: moped-editor/src/views/projects/projectView/ProjectWorkActivity/ProjectWorkActivityTable.js
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectWorkActivity/ProjectWorkActivityTable.js:190]: Could you delete these lines now that there is a new version of this function?

File: moped-editor/src/views/projects/projectView/ProjectWorkActivity/ProjectWorkActivityTable.js
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectWorkActivity/ProjectWorkActivityTable.js:201]: Just want to comment that the React tutorials served you well! `handleDeleteConfirmed` is an improved name for this function that [responds to an event](https://react.dev/learn/responding-to-events). 🚀

**Review by mddilley — CHANGES_REQUESTED** [review]
This works soo great!! I tested the modal locally, and I see the same networks requests happening to delete the work activity row and then fetch the updated data as before. Thanks for diving into this one and getting spun up so quickly! 🚀

I requested a few changes unrelated to the functionality, and they all revolve around smaller details that we didn't have time to squeeze in and talk over. I'll request the rest of the devs after these updates, and then we can 🚢. Thanks again!

**Comment by arkadeep** [comment]
Hey @mddilley, I think I got everything. I was confused about the changes in the package-lock file. I hope I didn't break anything.

File: moped-editor/package-lock.json
Comment (mddilley) [inline moped-editor/package-lock.json:1]: This is looking better! There are still a few lines that I think were generated by your code editor. [VSCode does something similar](https://code.visualstudio.com/docs/sourcecontrol/overview#_merge-conflicts) when resolving branch conflicts.

I think the most straightforward way to remove these changes is to copy/paste the contents of [this file in the main branch](https://github.com/cityofaustin/atd-moped/blob/main/moped-editor/package-lock.json) into this file your branch. Since you didn't make any changes to npm packages (libraries) then these changes can be discarded.

This goes way beyond the scope of this work, but you can read more about this file [here](https://docs.npmjs.com/cli/v10/configuring-npm/package-lock-json) if you are curious.

File: moped-editor/src/views/projects/projectView/ProjectWorkActivity/ProjectWorkActivityTable.js
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectWorkActivity/ProjectWorkActivityTable.js:265]: I missed this before, but could you update this `DeleteConfirmationModal` component to be [self-closing](https://react.dev/learn/writing-markup-with-jsx#2-close-all-the-tags)? It renders fine either way, but this is a convention for components that don't wrap other components and stand on their own.

**Comment by mddilley** [comment]
Thanks @arkadeep - I left two last comments for changes, and then we can get this tied up! I also missed going over our code formatting helper called [Prettier](https://prettier.io/). I can push one last commit to run that formatter on this code, but I wanted to let you know about another tool we use on the team.

All of this is so rad! Thank you again for all the work that you put into it! 🙌

File: moped-editor/src/views/projects/projectView/ProjectWorkActivity/ProjectWorkActivityTable.js
Comment (johnclary) [inline moped-editor/src/views/projects/projectView/ProjectWorkActivity/ProjectWorkActivityTable.js:192]: i noticed that down in the `onSubmitCallback` we use the `refetch().then(...)` pattern, and it had me wondering if we should use that here, so that we would wait to `setActivityToDelete()` until the refetch completes, but i do not really see any benefit in that. 

anyway, because `refetch()` returns a `promise` you should be able to chain it like this, if you wanted to:

```javascript
      deleteContract({ variables: { id: activityToDelete } })
        .then(() => refetch())
        .then(() => setActivityToDelete(null));
```

and—this is just an observation—looking at the DeleteConfirmationModal [code](https://github.com/cityofaustin/atd-moped/blob/49feca21a2b80ce53d5113735dd08b01725d8f6d/moped-editor/src/views/projects/projectView/DeleteConfirmationModal.js#L44-L47), it would be a nice future enhancement to make the `handleDeleteClose` callback wait until the `submitDelete` function completes—that would prevent the modal from closing until the mutation actually completes.

that's totally outside the scope of this work—my mind just went there when i was looking at the callback behavior in this PR.

**Review by johnclary — APPROVED** [review]
@arkadeep this looks great and works as promised. i left a meandering comment about the callback functions. this looks ready to ship to me—it's a nice little improvement to the work activity UX 

🚢🚢🚢🚢🚢🚢🚢🚢

**Review by roseeichelmann — APPROVED** [review]
its super nice to have these delete modals streamlined now! yay for writing your first react code :partying_face:  this looks good to go once you address @mddilley comments :))

File: moped-editor/src/views/projects/projectView/ProjectWorkActivity/ProjectWorkActivityTable.js
Comment (mddilley) [reply moped-editor/src/views/projects/projectView/ProjectWorkActivity/ProjectWorkActivityTable.js:192]: great catch - @johnclary! i'll create an spinoff snackoo issue to fix up `DeleteConfirmationModal` in this way later on. I think that will be a great enhancement too. 🙏

File: moped-editor/package-lock.json
Comment (arkadeep) [reply moped-editor/package-lock.json:1]: Done!

File: moped-editor/src/views/projects/projectView/ProjectWorkActivity/ProjectWorkActivityTable.js
Comment (arkadeep) [reply moped-editor/src/views/projects/projectView/ProjectWorkActivity/ProjectWorkActivityTable.js:265]: Done, thanks for the input!

Comment (mddilley) [reply moped-editor/src/views/projects/projectView/ProjectWorkActivity/ProjectWorkActivityTable.js]: awesome - thanks for updating!

**Comment by arkadeep** [comment]
@mddilley, thanks for all the input. I'm sorry for the package-lock mess up. I should have seen those changes. But I can assure you that it should be okay now!

**Comment by mddilley** [comment]
@arkadeep Thanks and no apologies needed. Seeing unexpected code changes when creating a pull request is all part of the fun! We see it happen all the time. Thanks for cleaning that up, and I'll approve this shortly. 

Thank you again for diving into React! 🤿 🎉

**Review by frankhereford — APPROVED** [review]
# 🚢🚢🚢 

Thanks Arkadeep! This works great and looks even better! 💯

**Review by mddilley — APPROVED** [review]
I tested this out in the work activity table again, and I see it working just as expected. 🚢 🚀

**Pull request merged**
